plugins/network-elements/netscaler/src/com/cloud/network/element/NetscalerElement.java

```diff
@@ -260,7 +264,7 @@ public boolean applyLBRules(Network config, List<LoadBalancingRule> rules) throw
         Map<Capability, String> lbCapabilities = new HashMap<Capability, String>();
 
         // Specifies that the RoundRobin and Leastconn algorithms are supported for load balancing rules
-        lbCapabilities.put(Capability.SupportedLBAlgorithms, "roundrobin,leastconn");
+        lbCapabilities.put(Capability.SupportedLBAlgorithms, "roundrobin,leastconn,source");
```

Comment:
The list of strings declared on this line must be sync with the list used to validate the algorithm on line 237.  Therefore, please consider extracting this list to a constant `ImmutableSet<String>`, `SUPPORTED_ALGORITHMS` in order to ensure cohesion between declaration and validation.  With this change, the validateLBRule can be re-implemented as follows:

```java
public boolean validateLBRule(Network network, LoadBalancingRule rule) {
    Preconditions.checkArgument(network != null, "validateLBRule requires a non-null network");
    Preconditions.checkArgument(rule != null, "validateLBRule requires a non-null rule");

    return (canHandle(network, Service.Lb)) : SUPPORTED_ALGORITHMS.contains(rule.getAlgorithm()) : true;
}
```

This line can be re-implemented as follows:

```java
lbCapabilities.put(Capability.SupportedLBAlgorithms, Joiner.on(",").join(SUPPORTED_ALGORITHMS));
```